Tests/TuistAutomationTests/Device/DeviceControllerTests.swift

```diff
@@ -79,11 +79,58 @@ final class DeviceControllerTests: TuistUnitTestCase {
                     platform: .iOS,
                     osVersion: "17.6.1"
                 ),
+                PhysicalDevice(
+                    id: "00008301-F856EAF4350DA92F",
+                    name: "My Watch",
+                    platform: .watchOS,
+                    osVersion: nil
+                ),
             ],
             got
         )
     }
 
+    func test_findAvailableDevices_when_list_decode_failed() async throws {
+        // Given
+        var devicesListOutputPath: AbsolutePath?
+
+        given(commandRunner)
+            .run(arguments: .any, environment: .any, workingDirectory: .any)
+            .willProduce { arguments, _, _ in
+                XCTAssertEqual(
+                    [
+                        "/usr/bin/xcrun", "devicectl",
+                        "list", "devices",
+                        "--json-output",
+                    ],
+                    arguments.dropLast()
+                )
+
+                devicesListOutputPath = try? arguments.last.map { try AbsolutePath(validating: $0) }
+
+                if let devicesListOutputPath {
+                    let semaphore = DispatchSemaphore(value: 0)
+                    Task {
+                        try await self.fileSystem.writeText("", at: devicesListOutputPath)
+                        semaphore.signal()
+                    }
+                    semaphore.wait()
+                }
+
+                return .init(
+                    unfolding: {
+                        nil
+                    }
+                )
```

Comment:
What do you think about writing a helper method to share the stubbing between the tests in this file?

Reply:
I have add a `write(text:at:)` method in `DeviceControllerTests.swift`. Do you think anything could be improved?

Reply:
No, that looks good!